### ISSUE TITLE
Typescript errors while compiling 

### ISSUE THREAD

**Issue by eckstaedt** [fullcalendar/fullcalendar#4035]
fulcalendar.d.ts version 3.8.1:
TS (2.6.2) compiler with strict null checking set is giving me errors because of wrong indexer type in this interface:

Line 1428:
```javascript
    export interface ButtonTextCompoundInput {
	    prev?: any;
	    next?: any;
	    prevYear?: any;
	    nextYear?: any;
	    today?: any;
	    month?: any;
	    week?: any;
	    day?: any;
	    [viewId: string]: string;
    }
```

```
Property 'prev' of type 'string | undefined' is not assignable to string index type 'string'.
```

The same error for all other properties of this interface.

**Comment by acerix** [comment]
I don't have any error with compilation, is it when you run `gulp dist` as described here?

https://github.com/fullcalendar/fullcalendar/blob/master/CONTRIBUTING.md#development-workflow

**Comment by eckstaedt** [comment]
No I simply start my application with npm. 
Using Webpack 3 and awesome-typescript-loader.

**Comment by arshaw** [comment]
this is the same problem reported in https://github.com/fullcalendar/fullcalendar/issues/3889#issuecomment-360055707

has to do with `strictNullChecks` being turned on.

i have this fixed in  master and will release soon

**Comment by eckstaedt** [comment]
sorry the code snippet was already the modified types from me. This is the right one. 

    export interface ButtonTextCompoundInput {
	    prev?: string;
	    next?: string;
	    prevYear?: string;
	    nextYear?: string;
	    today?: string;
	    month?: string;
	    week?: string;
	    day?: string;
	    [viewId: string]: string;
    }


Hope this not confusing...

**Comment by arshaw** [comment]
fixed in v3.8.2
